Library/Homebrew/formula_auditor.rb

```diff
@@ -235,8 +235,7 @@ def audit_deps
             problem "Dependency '#{dep.name}' was renamed; use new name '#{dep_f.name}'."
           end
 
-          if self.class.aliases.include?(dep.name) &&
-             dep_f.core_formula? && !dep_f.versioned_formula?
+          if self.class.aliases.include?(dep.name) && dep_f.core_formula?
```

Comment:
We're checking here whether the formula being depended on is a core formula but I think we should be checking for both the dependency and the main formula because depending on a versioned alias is actively desirable outside of homebrew/core. Thoughts?

Reply:
You're right! 
I've actually misread the condition, in my head exactly what you're saying 😄
I'll put this check after `next unless @core_tap`. And then we don't really need to check `dep_f.core_formula?` because formulae from homebrew-core can depend only on formulae from homebrew-core